lib/arelastic/nodes/node.rb

```diff
@@ -22,6 +24,26 @@ def read_option!(options, key)
       def ==(other)
         other.is_a?(Arelastic::Nodes::Node) && as_elastic == other.as_elastic
       end
+
+      def hash
+        as_elastic.hash
+      end
+
+      def eql?(other)
+        as_elastic.eql?(other.as_elastic)
+      end
+
+      def merge_expression(expr, klass:, merge_test_method:)
+        mergeable, other = expr.partition { |el| el.is_a?(klass) }
+        merged = mergeable.group_by(&merge_test_method).map do |merge_test_value, group|
+          if group.size > 1
+            klass.new(merge_test_value, group.map(&:query))
+          else
+            group
+          end
+        end.flatten
```

Comment:
can use `flat_map` instead of `map.flatten`

Reply:
Fixed